### DISCUSSION TITLE
Can the Wiki be moved to a repository?

### DISCUSSION THREAD

**Discussion by inobelar** [crsf-wg/crsf#15] [General]
First of all, I want to thank you for creating this documentation repository, it is very helpful!

I'm developing a parser/generator of CRSF frames, and I want to participate in updating the documentation. For example, I noticed that in [CRSF_FRAMETYPE_ATTITUDE](https://github.com/crsf-wg/crsf/wiki/CRSF_FRAMETYPE_ATTITUDE) missed **endianess** (pitch/yaw/roll packed as big-endian, as I can see in [betaflight sources](https://github.com/betaflight/betaflight/blob/d82c8facb22fac637db8cdb28904875f713a0f87/src/main/telemetry/crsf.c#L370-L377)). 

First of all, currently the wiki can't be edited :disappointed:  But also (even if it were open to everyone), I don't want to introduce unnecessary/incorrect information - my edits may be unacceptable for one reason or another, or redundant. So I'm asking - could you please upload the 'wiki repository', and put it in the 'main repository'? - So that edits undergo review (via a pull request), and are not available without verification (as in current wiki).

Also, I would like to add links to the use of frames (structure, packing/unpacking) in different projects - iNav/BetaFlight/ArduPilot and examples of frames (in raw bytes - so that anyone can check serialization/deserialization without hardware, in any programming language).

And also I want to reformat the package descriptions to a 'tabular' form, in the following style (for example):

----

# UAS Attitude frame

- Type: Hex: `0x1E` / Dec: `30`
- Size: 10 bytes
- Telemetry frame: +
- Extended  frame: -
- Transmission: Flight Controller --> Receiver

| name | type | endianess | description |
| :-- | :-- | :-- | :--
| Pitch | int16_t | big-endian | angle in radians / 10000
| Roll | int16_t | big-endian | angle in radians / 10000
| Yaw | int16_t | big-endian | angle in radians / 10000

> [!NOTE]  
> Pitch/Roll/Yaw angles are sent in `radians/10000`, so:
> - 180 degrees sent as `0x7AB7` / `31415`
> - 45 degrees sent as `0x1EAD` / `7853`
> - -45 degrees sent as `0xE153` / `-7853`

> [!WARNING]
> All values must be in the +/-180 degree +/-PI radian range.

## Example frame

- `0xC8, 0x08, 0x1E, 0x02, 0x85, 0x00, 0xF4, 0x79, 0x6C, 0x7B` (10 bytes)
  - Sync byte: `0xC8`
  - Length: `0x08` / `8`
  - Type: `0x1E` / `30`
  - Payload: `0x02, 0x85, 0x00, 0xF4, 0x79, 0x6C` (6 bytes)
    | name | bytes (big-endian) / dec | int16_t (little-endian) hex / dec | radians / degrees |
    | :-- | :-- | :-- | :-- |
    | Pitch | `0x02, 0x85` / `-31486`  | `0x8502` / `645` | `0,064500` / `3,695578`
    | Roll | `0x00, 0xF4` / `-3072`| `0xF400` / `244` | `0,024400` / `1,398017`
    | Yaw | `0x79, 0x6C` / `27769` | `0x6C79` / `31084` | `3,108400` / `178,098207`
  - CRC: `0x7B`
  
## Usage

- BetaFlight
  - [serialization](https://github.com/betaflight/betaflight/blob/d82c8facb22fac637db8cdb28904875f713a0f87/src/main/telemetry/crsf.c#L370-L377)
- ArduPilot
  - [structure](https://github.com/ArduPilot/ardupilot/blob/247f5e54460b7b9d8046ee508bb886c2e761cd27/libraries/AP_RCTelemetry/AP_CRSF_Telem.h#L102-L106)
  - [serialization](https://github.com/ArduPilot/ardupilot/blob/247f5e54460b7b9d8046ee508bb886c2e761cd27/libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp#L1014-L1029)

**Comment by CapnBry** [comment]
It can not be moved to a repo because then it would require me to start reviewing PRs and explaining everything to everyone all the time. Sort of like how I now have to examine this for like an hour to make sure it is accurate 😅 The amount of burden on me you're suggesting is too much for the minimal chance that someone might provide an accurate edit or improvement. I do agree that the formatting could be better but also this is pretty low priority for me because I feel like I'm the only person who ever looks at it.

All values are big-endian except where otherwise noted.
